### ISSUE TITLE
Seaglass crashes when "Homeserver" UserDefault is set to an invalid URL 

### ISSUE THREAD

**Issue by aaronraimist** [neilalexander/seaglass#118]
- [ ] User's shouldn't be able to enter an invalid URL using the interface
- [ ] If they do set an invalid URL, Seaglass shouldn't crash

@peter2121 solved it by changing 
`let homeserver = URL(string: defaults.string(forKey: "Homeserver") ?? "https://matrix.org")!.host ?? "matrix.org"`
to
`let homeserver = URL(string: defaults.string(forKey: "Homeserver") ?? "https://matrix.org")?.host ?? "matrix.org"`

in `LoginViewController.swift`

**Comment by michaelsmoody** [comment]
I suspect this is what I'm experiencing right now. I attempted to change the home server today, due to the matrix.org issue, and when I removed the custom homeserver entry, and restarted Seaglass, it now crashes on startup. Platform is macOS 10.14.3, with install via brew. I'm not sure where the "settings" are stored on macOS to remove them, as reinstalling the application does not fix the crash.
